### ISSUE TITLE
GPS lost during the activity

### ISSUE THREAD

**Issue by acca** [11samype/CIQBitcoinWidget#1]
It seems that if used during an activity (let's say run or walk) it disable the GPS for a while. If I switch back to the activity screen after a while I see a message saying the GPS is back online. My watch is Forerunner 245.

I can help with debugging/testing.

**Comment by 11samype** [comment]
Thank you for bringing this to my attention. I tried to reproduce on a vivoactive 3 and didn't run into the issue. I do have a FR 245M but don't know of a way to get to the widgets while in an activity. Are you accessing the widgets by holding the down button? That shows the music controls on my FR 245M.

Honestly even if I can reproduce, I don't know what can be done to mitigate the issue. This app is very simple, and does not do anything with GPS. Only does a network request and a draw call to the screen.

I might be able to get my hands on a regular 245 if I can't find a way to show widgets on a 245M.

**Comment by acca** [comment]
Nice to hear you. I can access the widget during an activity because I disabled the "Lap button" in the activity settings so my "back" button simply bring me to the main watchface, from there I can move to the widget.

I perfectly understand that the widget is so simple and should not interfere with the GPS but based on my quick tests this is the case. I very appreciate if you can give it a try on your side to confirm it or not.

**Comment by 11samype** [comment]
Right, I forgot about disabling the lap button. I am able to reproduce the issue on my 245M. I will try to diagnose the problem this weekend.

**Comment by 11samype** [comment]
I did some testing over the weekend and found that any CIQ that I tried caused GPS to cut out. Built in widgets do not exhibit the issue. I wonder if it is some issue with the 245 firmware. @acca can you confirm if you see the same behavior with other CIQ widgets? I will submit a bug report to Garmin.

**Comment by acca** [comment]
Unfortunately I don't have anymore the Forerunner 245. I replaced it with the Fenix 6 and I can tell you that in the Fenix I don't see such issue. I could test it a little bit more on the Fenix if you like.
